repository/repository.go

```diff
@@ -111,7 +115,9 @@ func (r *Repository) LoadBlob(t pack.BlobType, id backend.ID) ([]byte, error) {
 		return nil, err
 	}
 
-	buf, err := ioutil.ReadAll(rd)
+	// make buffer that is large enough for the complete blob
+	cbuf := make([]byte, length)
```

Comment:
can you think of a more descriptive variable name than `cbuf`? I'm assuming this is a buffer that holds ciphertext? how about `ciphertextBuf`?